cpp/find_latest_journal_year.cc

```diff
@@ -157,12 +157,28 @@ void ProcessRecords(MARC::Reader * const marc_reader, std::unordered_map<std::st
             continue;
 
         std::string year_as_string;
-        for (const auto &field : record.getTagRange("936")) {
-            const MARC::Subfields subfields(field.getSubfields());
-            year_as_string = subfields.getFirstSubfieldWithCode('j');
-            if (not year_as_string.empty())
-                break;
+
+        std::vector<std::string> subfields;
```

Comment:
- `subfields` is a bad variable name here because it will not contain entire subfields. instead it contains splitted parts of a subfield value, so you should find a better name
- variables should always be declared at the latest possible point. if you compare this with filtered_dates, that one is valid during the whole loop so it is ok to declare filtered_dates before the loop, but since subfields will be reset when calling StringUtil::Split() you should move the declaration next to that line.